sql/wsrep_encryption.h

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2019 Codership Oy <info@codership.com>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#ifndef WSREP_ENCRYPTION_H
+#define WSREP_ENCRYPTION_H
+
+#include "wsrep/encryption_service.hpp"
+#include <vector>
+
+class Wsrep_encryption_service : public wsrep::encryption_service
+{
+public:
+
+  static Wsrep_encryption_service* instance()
+  {
+    if (m_instance == 0)
+    {
+      m_instance = new Wsrep_encryption_service();
```

Comment:
We don't need to allocate this dynamically as the constructor does not take any arguments,

Reply:
Singleton is not needed anymore, as we don't set encryption key through class.